### ISSUE TITLE
Progmatically clearing backend cache shows empty mempool for a while

### ISSUE THREAD

**Issue by wiz** [mempool/mempool#3221]
After clearing backend cache the mempool incorrectly appears empty until the cache can be re-populated.

Instead of displaying incorrect data, we should not show anything in the projected mempool blocks, or log any historical data into the database, etc. similar to how we return HTTP 502 for critical API endpoints such as fee estimations until the mempool is "in sync" again.

<img width="1260" alt="Screenshot 2023-03-04 at 15 18 51" src="https://user-images.githubusercontent.com/232186/222879579-34051f0a-2931-4428-b14e-fe6cd4bff1ec.png">

<img width="553" alt="Screenshot 2023-03-04 at 15 15 19" src="https://user-images.githubusercontent.com/232186/222879520-63e1298c-7cc2-476e-956c-ce34823b8ceb.png">

**Comment by nymkappa** [comment]
I don't think it's related to clearing the cache. I think that's a bug that always has been here pretty much. It's only clearly visible now because we cleared the cache.

@softsimon what do you think?

**Comment by softsimon** [comment]
2nd bug, we already prevent graph entries before the mempool is fully in sync. So no idea what went on here.

**Comment by wiz** [comment]
```
root@localhost [mempool]> select added, vsize_1 from statistics order by added desc limit 200;
+---------------------+----------+
| added               | vsize_1  |
+---------------------+----------+
...
| 2023-03-04 05:43:35 | 38492371 |
| 2023-03-04 05:42:35 | 38492269 |
| 2023-03-04 05:41:35 | 38491955 |
| 2023-03-04 05:40:35 | 38495283 |
| 2023-03-04 05:39:35 | 38494186 |
| 2023-03-04 05:38:35 | 38494329 |
| 2023-03-04 05:37:35 | 26356227 | <--- here
| 2023-03-04 05:36:35 | 38494050 |
| 2023-03-04 05:35:35 | 38493774 |
| 2023-03-04 05:34:35 | 38493681 |
| 2023-03-04 05:33:35 | 38493571 |
| 2023-03-04 05:32:35 | 38493428 |
| 2023-03-04 05:31:35 | 38493572 |
| 2023-03-04 05:30:35 | 38493428 |
| 2023-03-04 05:29:35 | 38485853 |
| 2023-03-04 05:28:34 | 38491781 |
| 2023-03-04 05:27:34 | 38485159 |
| 2023-03-04 05:26:34 | 38485068 |
| 2023-03-04 05:25:34 | 38485094 |
| 2023-03-04 05:24:34 | 38484259 |
| 2023-03-04 05:23:34 | 38484259 |
| 2023-03-04 05:22:34 | 38484157 |
| 2023-03-04 05:21:34 | 38484003 |
| 2023-03-04 05:20:34 | 38483675 |
| 2023-03-04 05:19:34 | 38483540 |
| 2023-03-04 05:18:34 | 38483438 |
| 2023-03-04 05:17:34 | 38482207 |
| 2023-03-04 05:16:34 | 38482053 |
| 2023-03-04 05:15:33 | 38482053 |
| 2023-03-04 05:14:33 | 38474837 |
+---------------------+----------+
200 rows in set (0.001 sec)
```

**Comment by nymkappa** [comment]
My suggestion would be to compare Bitcoin Core mempool size and Esplora mempool size as an additional condition for triggering the mempool protection. If the gap between both is too big (and their both fully synced), then we can trigger the protection. @softsimon

**Comment by softsimon** [comment]
> My suggestion would be to compare Bitcoin Core mempool size and Esplora mempool size as an additional condition for triggering the mempool protection. If the gap between both is too big (and their both fully synced), then we can trigger the protection. @softsimon

Good idea, but note also that the mempool protection was put in place to handle Bitcoin Core restarts in the first place, not Esplora.

**Comment by nymkappa** [comment]
@wiz @softsimon is this still a things ?

**Comment by wiz** [comment]
I think it's fine for prod now that we are using redis but it might still reproduce using disk cache for community nodes on RPI devices?
